internal/runtime/hcsv2/workload_container.go

```diff
@@ -163,8 +167,67 @@ func setupWorkloadContainerSpec(ctx context.Context, sbid, id string, spec *oci.
 	// Force the parent cgroup into our /containers root
 	spec.Linux.CgroupsPath = "/containers/" + id
 
+	if spec.Windows != nil && len(spec.Windows.Devices) > 0 {
```

Comment:
Just do this check in `addNvidiaDevicePreHook`

Reply:
I feel like that change would make the code a bit less understandable here, since we aren't always wanting to add an nvidia prestart hook. Do you feel strongly about this?

Reply:
I'm saying that len(spec.Windows.Devices) is not a good enough check. So its an odd pre-check. I get the spec.Windows != nil part thats obvious. But really there needs to be spec.Windows.Device[i] where IDType == "gpu" so seems weird to have this check here. Alternatively do the loop iteration here and then call the perHook when you find a matching device so that its more obvious.

Reply:
@jterry75 is the new check okay?